### PR TITLE
[Merged by Bors] - chore: backports for leanprover/lean4#4814 (part 20)

### PR DESCRIPTION
**Pull request by kim-em**
see https://github.com/leanprover-community/mathlib4/pull/15245

### CODE REVIEW

File: Mathlib/Analysis/Normed/Operator/LinearIsometry.lean [hunk 1]
```diff
@@ -170,14 +170,6 @@ theorem ext {f g : E →ₛₗᵢ[σ₁₂] E₂} (h : ∀ x, f x = g x) : f = g
 
 variable [FunLike 𝓕 E E₂]
 
-protected theorem congr_arg {f : 𝓕} :
-    ∀ {x x' : E}, x = x' → f x = f x'
-  | _, _, rfl => rfl
-
-protected theorem congr_fun {f g : 𝓕} (h : f = g) (x : E) :
-    f x = g x :=
-  h ▸ rfl
```
Comment (sgouezel) [inline Mathlib/Analysis/Normed/Operator/LinearIsometry.lean:179]: Why did you remove these ones?

File: Mathlib/RingTheory/IntegralClosure/IsIntegral/Basic.lean [hunk 2]
```diff
@@ -47,9 +47,9 @@ theorem IsIntegral.map {B C F : Type*} [Ring B] [Ring C] [Algebra R B] [Algebra
 section
 
 variable {A B : Type*} [Ring A] [Ring B] [Algebra R A] [Algebra R B]
-variable (f : A →ₐ[R] B) (hf : Function.Injective f)
 
-theorem isIntegral_algHom_iff {x : A} : IsIntegral R (f x) ↔ IsIntegral R x := by
+theorem isIntegral_algHom_iff (f : A →ₐ[R] B)  (hf : Function.Injective f) {x : A} :
```
Comment (sgouezel) [inline Mathlib/RingTheory/IntegralClosure/IsIntegral/Basic.lean:51]:
```suggestion
theorem isIntegral_algHom_iff (f : A →ₐ[R] B) (hf : Function.Injective f) {x : A} :
```

File: Mathlib/RingTheory/IntegralClosure/IsIntegral/Basic.lean [hunk 3]
```diff
@@ -87,18 +87,18 @@ variable (f : R →+* B)
 theorem RingHom.isIntegralElem_zero : f.IsIntegralElem 0 :=
   f.map_zero ▸ f.isIntegralElem_map
 
-theorem isIntegral_zero : IsIntegral R (0 : B) :=
+theorem isIntegral_zero [Algebra R B]  : IsIntegral R (0 : B) :=
```
Comment (sgouezel) [inline Mathlib/RingTheory/IntegralClosure/IsIntegral/Basic.lean:90]:
```suggestion
theorem isIntegral_zero [Algebra R B] : IsIntegral R (0 : B) :=
```

File: Mathlib/RingTheory/IntegralClosure/IsIntegral/Basic.lean [hunk 4]
```diff
@@ -87,18 +87,18 @@ variable (f : R →+* B)
...
   f.map_one ▸ f.isIntegralElem_map
 
-theorem isIntegral_one : IsIntegral R (1 : B) :=
+theorem isIntegral_one [Algebra R B] : IsIntegral R (1 : B) :=
   (algebraMap R B).isIntegralElem_one
 
 variable (f : R →+* S)
 
-theorem IsIntegral.of_pow {x : B} {n : ℕ} (hn : 0 < n) (hx : IsIntegral R <| x ^ n) :
+theorem IsIntegral.of_pow [Algebra R B]  {x : B} {n : ℕ} (hn : 0 < n) (hx : IsIntegral R <| x ^ n) :
```
Comment (sgouezel) [inline Mathlib/RingTheory/IntegralClosure/IsIntegral/Basic.lean:101]:
```suggestion
theorem IsIntegral.of_pow [Algebra R B] {x : B} {n : ℕ} (hn : 0 < n) (hx : IsIntegral R <| x ^ n) :
```

**Review by sgouezel — COMMENTED** [review]
bors d+

File: Mathlib/Analysis/Normed/Operator/LinearIsometry.lean (hunk 1 above)
Comment (semorrison) [reply Mathlib/Analysis/Normed/Operator/LinearIsometry.lean:179]: Hmm, they were unused, inconvenient to move to a place where the variables would be suitable, and hopefully are taken care of by `FunLike.congr_arg`. I could go through the deprecation hoops, but I'm willing to cut a few corners getting this project out of the way. :-)

Comment (sgouezel) [reply Mathlib/Analysis/Normed/Operator/LinearIsometry.lean:179]: ok, fine for these!

**Comment by semorrison** [comment]
bors merge
